sycl/doc/extensions/proposed/sycl_ext_codeplay_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,421 @@
+local memory. Typically addresses which reside in the local memory of a 
+work-group can only be accessed by the work-items of that work-group. 
+Therefore, to access another work-group's local memory, the address needs to be
+mapped such that the address in another work-group is accessible within the 
+calling work-item. Further, to access another another work-group's local memory,
+all the work-groups within the cluster must exist and the work-groups should
+not cease to exist before all the memory operations are completed. This can be 
+ensured by synchronizing all the work-items within the cluster before and after
+the local memory operations, using `group_barrier`.
+
+A member function of the `cluster_group` class; 
+`map_cluster_local_pointer` will perform the mapping and return a pointer 
+which can then be dereferenced by the calling work-item. 
+
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|T*  map_cluster_local_pointer(T* addr, size_t group_id)
+|Accepts the equivalent address to the memory location relative to the calling 
+work-item which is to be mapped from the local memory of the work-group, as 
+specified by the group_id, denoting the linear group_id within the cluster
+|===
+
+Conversely, `get_linear_cluster_group_id_for_local_pointer` will return the 
+linearized id of the work-group a mapped local memory address belongs to.
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|size_t  get_linear_cluster_group_id_for_local_pointer(T* addr)
+|Accepts a mapped pointer to the local memory of another work-group and returns
+the linearized id of the work-group in the cluster from whom's local memory the
+mapped pointer was obtained
+|===
+
+
+== Cluster Memory Fence Scope and Barrier
+
+Work-items in a work-group can access a local memory address from another
+work-group in the cluster-group, which has been mapped as described above. To
+facilitate this, a new memory scope is introduced to the `memory_scope` class;
+`ext_codeplay_cuda_cluster_group` which indicates a memory ordering
+constraint that applies to all work-items in the same cluster-group. This memory
+scope can be used with `atomic_ref`` and other SYCL APIs that use
+`memory_scope`.
+
+[source, c++]
+----
+namespace sycl {
+
+  enum class memory_scope {
+  ...
+  ext_codeplay_cuda_cluster_group,
+  ...
+  };
+
+  namespace ext::codeplay::experimental::cuda {
+    inline constexpr auto memory_scope_cluster_group
+      = memory_scope::ext_codeplay_cuda_cluster_group;
+  } // namespace ext::codeplay::experimental::cuda
+
+----
+
+namespace ext::codeplay::experimental::cuda {
+inline constexpr auto memory_scope_cluster_group = memory_scope::ext_codeplay_cuda_cluster_group;
+} // namespace ext::codeplay::experimental::cuda
+
+
+[source,c++]
+----
+    class cluster_group {
+    public:
+    ...
+    static constexpr memory_scope fence_scope = 
+                memory_scope::ext_codeplay_cuda_cluster_fence;
```

Comment:
Personally, I find it confusing to add this separate code listing that shows the `fence_scope` member type alias.  I'd add this in the main synopsis for `cluster_group` above, so that there is a single synopsis that shows all the public members.

Reply:
Done

Reply:
I don't see any change to the example.  Maybe you forgot to push a commit?